src/uu/ls/src/ls.rs

```diff
@@ -142,14 +142,16 @@ const DEFAULT_TERM_WIDTH: u16 = 80;
 #[derive(Debug)]
 enum LsError {
     InvalidLineWidth(String),
-    NoMetadata(PathBuf),
+    IOError(std::io::Error),
```

Comment:
Thank you for your review.  I popped an error in the test after naively trying UIoError.  

> thread 'test_ls::test_ls_files_dirs' panicked at ''ls: cannot access 'doesntexist': Entity not found ' does not contain ''doesntexist': No such file or directory'', tests/common/util.rs:396:9

"Entity not found" does not match GNU error codes.  I think it should be "No such file or directory." I'm more than pleased to fix myself but do you have a utility which you would consider uses best practices, so I can implement according to the preferred approach?

Reply:
So, what I would do is to make this variant `IOError(UIoError)` and implement `From<UIoError>` for `LsError`. Something like this:
```rust
let unwrapped = match entry {
    Ok(path) => path,
    Err(error) => {
        let _ = out.flush();
        show!(LsError::IOError(error));
        break;
    }
};
```
then becomes something like this:
```rust
use uucore::{error::FromIo, display::Quotable};
...
let unwrapped = entry.map_err_context(|| name.maybe_quote().to_string())?;
```

Surprisingly, there's no util that does exactly this. But [`install`](https://github.com/uutils/coreutils/blob/master/src/uu/install/src/install.rs) has a slightly different approach, which is also quite nice.

Reply:
The "Entity not found" error is interesting. If you run into it again, please show me how it happens, because that would be a bug. UIoError should already show the "No such file or directory" message instead of "Entity not found"

Reply:
I think you may have a bug then.  I just pushed my naive implementation.  Let me see if I can point you to it.